channel/signedstate.go

```diff
@@ -1,17 +1,60 @@
 package channel
 
 import (
+	"errors"
+
 	"github.com/statechannels/go-nitro/channel/state"
 )
 
-type SignedState struct {
-	State state.VariablePart
-	Sigs  map[uint]state.Signature // keyed by participant index
+type signedState struct {
+	state state.State
+	sigs  map[uint]state.Signature // keyed by participant index
+}
```

Comment:
I understand the motivation to make these data private, but I also spot that you have made the type a non-exported type -- so it is not available outside of the `channel` package. Was that intentional?

Reply:
I think probably this type wants to live in the `state` package and be exported by that package for use here? Otherwise the public / private distinction doesn't make a heap of sense.

Reply:
> I understand the motivation to make these data private, but I also spot that you have made the type a non-exported type -- so it is not available outside of the channel package. Was that intentional?

Yep although perhaps that's not the correct approach! By making `signedState` a private struct I was trying to prevent it from being instantiated or created outside of the `channel` package. Code external to the `channel` package can still call `State()` or `HasSignatures` but cannot create their own `signedState`
![image](https://user-images.githubusercontent.com/1620336/151598547-f32aad92-1f6f-4806-b084-a5c272a3df8c.png)

My assumption was that `signedState` is a helper structure within `channel`. The `channel` package would be the only one creating `signedState`s and external code would only be reading from the `signedState`, never modifying them.  To follow that pattern maybe there should be a signed state interface that is returned by `SignedStateForTurnNum` ?

It also depends on what the role of `SignedState` is. We could move it to a the `state` package and start accepting it in `func (c *Channel) AddSignedState` (instead of just a state and signature).

Reply:
I see! Let's leave it as is for now, we can always circle back to it if we want to later on.